rust/src/coordinator.rs

```diff
@@ -290,20 +339,26 @@ impl Coordinator {
     /// Transition to the next phase if the protocol conditions are satisfied.
     pub fn try_phase_transition(&mut self) {
         match self.phase {
-            Phase::Idle => self.proceed_sum_phase(),
+            Phase::Idle => {
+                self.proceed_sum_phase();
+                self.try_phase_transition();
```

Comment:
In the idle case it's necessary otherwise the state machine is stuck, unless someone manually call `try_phase_transition()`. I'm actually not sure that `Idle` state is needed, but refactoring this out is out of scope of this PR.

For the other states, I'm less sure it is needed, but in any case it doesn't hurt.